### PR TITLE
updates

### PR DESCRIPTION
**Pull request by zakird**

### CODE REVIEW

**Comment by zakird** [comment]
DO NOT MERGE

File: ztag/schema.py [hunk 1]
```diff
@@ -35,20 +35,22 @@ class CensysString(WhitespaceAnalyzedString):
...
     "ip_addresses":ListOf(IPAddress()),
     "directory_names":ListOf(zgrab_subj_issuer),
-    "edi_party_names":ListOf(SubRecord({
-        "name_assigner":CensysString,
-        "party_name":CensysString,
-    })),
+    "edi_party_names":ListOf(edi_party_name),
     "other_names":ListOf(SubRecord({
-        "id":String(),
+        "id":String(), # XXX is this an OID?
```
Comment (dadrian) [inline ztag/schema.py]: Yes, this is an OID. https://github.com/zmap/zgrab/blob/ad824319f6286ff5fbfe85b4605a672d8c13ba15/ztools/x509/pkix/json.go#L171

File: ztag/schema.py [hunk 2]
```diff
@@ -107,18 +109,31 @@ class CensysString(WhitespaceAnalyzedString):
...
+})
+
 zgrab_parsed_certificate = SubRecord({
     "subject":zgrab_subj_issuer,
     "subject_dn":CensysString(),
     "issuer":zgrab_subj_issuer,
     "issuer_dn":CensysString(),
     "version":Unsigned8BitInteger(),
-    "serial_number":String(doc="Serial number as an unsigned decimal integer. "\
+    "serial_number":String(doc="Serial number as an signed decimal integer. "\
```
Comment (dadrian) [inline ztag/schema.py]: 👏

File: ztag/schema.py [hunk 3]
```diff
@@ -186,25 +201,53 @@ class CensysString(WhitespaceAnalyzedString):
...
         "subject_alt_name":alternate_name,
         "issuer_alt_name":alternate_name,
-        "crl_distribution_points":ListOf(URI()),
+        "crl_distribution_points":ListOf(URL()),
         "authority_key_id":HexString(),
         "subject_key_id":HexString(),
-        "extended_key_usage":ListOf(Signed32BitInteger()), # TODO: what sized integer should this be?
-        "certificate_policies":ListOf(CensysString()),
+        "extended_key_usage":SubRecord({
+            "value":ListOf(Signed32BitInteger()), # TODO: what sized integer should this be?
```
Comment (dadrian) [inline ztag/schema.py]: There's only 9 defined in the RFC, but Golang parses them as a 32-bit signed integer.

File: ztag/schema.py [hunk 4]
```diff
@@ -186,25 +201,53 @@ class CensysString(WhitespaceAnalyzedString):
...
+            "server_auth":Boolean(doc="TLS WWW server authentication"),
+            "client_auth":Boolean(doc="TLS WWW client authentication"),
+            "code_signing":Boolean(doc="Signing of downloadable executable code"),
+            "email_protection":Boolean(doc="Email protection"),
+            "time_stamping":Boolean(doc="Binding the hash of an object to a time"),
+            "ocsp_signing":Boolean(doc="Signing OCSP responses")
```
Comment (dadrian) [inline ztag/schema.py]: These tenses aren't all the same in the docstrings, but I'm bad at English and can't tell you what's correct. Probably something authoritative?

File: ztag/schema.py (hunk 3 above)
Comment (zakird) [inline ztag/schema.py]: I say let's just leave it then.

File: ztag/schema.py (hunk 4 above)
Comment (zakird) [inline ztag/schema.py]: These are copy pasted from the RFC.

**Pull request merged**